docs/developer-guide.md

```diff
@@ -11,6 +11,45 @@ Build and test this project by running `build.ps1` on Windows, or by running `bu
 
 You can use [Visual Studio](https://www.visualstudio.com/vs/) or [Visual Studio Code](https://code.visualstudio.com/) with [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) to develop the project.
 
+## Local Debug Tutorial for Docfx v3
+
+> **For Microsoft Internal Users:**
+>
+> Make sure you have the permissions to run this repo locally. Contact [docfxvnext@microsoft.com](docfxvnext@microsoft.com) to add read permission, please.
+
+### Step 1: Clone Repo
+Clone the repo and check out to v3 branch.
+
+```shell
+git clone https://github.com/dotnet/docfx
+git checkout v3
+```
+### Step 2: Open the Repo with Visual Studio
+
+Open `docfx.sln` with Visual Studio. Add `Debug arguments` for `docfx` project to specify the repo you want to debug, build arguments for example: `build "C:\workspace\test-repo"`
+
+```shell
+build "your repo on local disk"
+```
+
+### Step 3: Debug
+
+Now you can set breakpoints and debug with your specified repo.
+
+
+### More Build Options:
```

Comment:
- A full cmd sample here would be clearer. e.g. `build "E:/docs repos/dataexplorer-docs-pr" --verbose -o "C:\Users\mingwli\Desktop\temp debug\output" --log "C:\Users\mingwli\Desktop\temp debug\output\errors.log" --output-type "pagejson"`

- may add some environment variables, e.g. config `DOCS_ENVIRONMENT` to point to `PPE`

Reply:
- Add some common scenarios as examples. 
- add environment variable description